cmd/client/main.go

```diff
@@ -52,7 +52,7 @@ var (
 	tokenPath          = kingpin.Flag("token-path", "Uses an OAuth 2.0 Bearer token found in this path to make scrape requests").String()
 	insecureSkipVerify = kingpin.Flag("insecure-skip-verify", "Disable SSL security checks for client").Default("false").Bool()
 	useLocalhost       = kingpin.Flag("use-localhost", "Use 127.0.0.1 to scrape metrics instead of FQDN").Default("false").Bool()
-	allowPort          = kingpin.Flag("allow-port", "Restricts the proxy to only being allowed to scrape the given port").Default("*").String()
+	allowPort          = kingpin.Flag("allow-port", "Restricts the proxy to only being allowed to scrape the given ports, separated by commas").Default("*").String()
```

Comment:
I think it is bad form to add parsing strategies to existing strings. It is more natural to add `allowPorts` as an array (or slice if you prefer golang terms)

Reply:
@richardelling Thank you for the feedback! My goal with this change is to keep everything backward compatible while adding the flexibility to specify multiple ports using a comma-separated list. This way, existing setups remain unchanged, but we gain more options for new configurations

Reply:
I'd prefer if this used [kingping.Strings()](https://pkg.go.dev/gopkg.in/alecthomas/kingpin.v2#FlagClause.Strings) as well. In this case an empty slice would be equivalent to `*`. Keeping the flag as `--allow-port` will keep backward compatibility, but should rename the `allowPort` var to `allowPorts`.